source/Classes/1.DscResources/08.nxScript.ps1

```diff
@@ -0,0 +1,118 @@
+[DscResource()]
+class nxScript
+{
+    [DscProperty()]
+    [System.String] $GetScript
+
+    [DscProperty(Key)]
+    [System.String] $TestScript
+
+    [DscProperty()]
+    [System.String] $SetScript
+
+    [DscProperty()]
+    [Reason[]] $Reasons
+
+    [nxScript] Get()
+    {
+        if (-not $this.GetScript)
+        {
+            # The GetScript script block was not defined
+            return $this
```

Comment:
$this.GetScript is not a boolean. I know PowerShell lets you do this when you are not in strict mode because null is falsey in PS but I think it should be `[string]::IsNullOrEmpty($GetScript)`?
Also can we make sure to populate the Reasons key with an empty array here? This may be returning null Reasons which means our agent will try to populate the Reasons with gibberish

Reply:
If I set $this.Reasons = @(), I get an error but if I set it to $null it's fine. Can we discuss tomorrow?

![image](https://github.com/Azure/nxtools/assets/6396451/bae78e98-3a55-4d29-b563-4fdb4b0130aa)

Reply:
Tried casting the empty array but I get same error.

$this.Reasons = [Reason[]]@()
or
$this.Reasons = [System.Collections.Generic.List[Reason]]@()

Now I am returning a default Reason: "The GetScript script block was no defined". Let me know what you think.